### PR TITLE
Make lookback duration configurable

### PR DESCRIPTION
**Pull request by benraskin92**

### CODE REVIEW

File: src/query/models/params.go [hunk 1]
```diff
@@ -34,10 +34,10 @@ const (
...
 )
 
-// LookbackDelta determines the time since the last sample after which a time
-// series is considered stale (inclusive).
-// TODO: Make this configurable
-var LookbackDelta = time.Minute
+// // LookbackDelta determines the time since the last sample after which a time
+// // series is considered stale (inclusive).
+// // TODO: Make this configurable
+// var LookbackDelta = time.Minute
```
Comment (benraskin92) [inline src/query/models/params.go]: Remove this commented out code

File: src/query/server/server.go [hunk 2]
```diff
@@ -77,6 +77,7 @@ var (
 	}
 
 	defaultCarbonIngesterWorkerPoolSize = 1024
+	defaultLookbackDuration             = 5 * time.Minute
```
Comment (benraskin92) [inline src/query/server/server.go]: This is what Prom uses

File: src/query/server/server.go [hunk 3]
```diff
@@ -162,6 +163,10 @@ func Run(runOpts RunOptions) {
 		logger.Fatal("could not create tag options", zap.Error(err))
 	}
 
+	if cfg.LookbackDuration == nil {
+		cfg.LookbackDuration = &defaultLookbackDuration
```
Comment (arnikola) [inline src/query/server/server.go]: This should be validated to be >= 0

Comment (benraskin92) [reply src/query/server/server.go]: Good call

Comment (arnikola) [reply src/query/server/server.go]: nit: Might be better to put this check on a validate function, but not really a big deal

Comment (robskillington) [reply src/query/server/server.go]: Can you add this as a method on the config object?

i.e.
```go
func (c Configuration) LookbackDurationOrDefault() (time.Duration, error) {
  if c.LookbackDuration == nil {
    return DefaultLookbackDuration, nil
  }
  v := *c.LookbackDuration
  if v < 0 {
    return nil, anError
  }
  return v, nil
}
```

File: src/query/server/server.go [hunk 4]
```diff
@@ -173,11 +172,11 @@ func Run(runOpts RunOptions) {
...
 
-	if cfg.LookbackDuration == nil {
-		cfg.LookbackDuration = &defaultLookbackDuration
-	} else if *cfg.LookbackDuration < 0 {
-		logger.Fatal("lookbackDuration must be > 0")
+	lookbackDuration, err := cfg.LookbackDurationOrDefault()
+	if err != nil {
+		logger.Fatal("error validating LookbackDuration", zap.Error(err))
 	}
+	cfg.LookbackDuration = &lookbackDuration
```
Comment (arnikola) [inline src/query/server/server.go]: nit newline

**Pull request merged**